customer_service/src/Controllers/CustomerController.php

```diff
@@ -0,0 +1,312 @@
+<?php
+namespace Customer\Controllers;
+
+use Lib\ResponseBuilder\JsonResponseBuilder;
+use Lib\ResponseBuilder\JsonClientResponseData;
+use Lib\ResponseBuilder\JsonClientResponseArray;
+use Customer\Services\CustomerService;
+use Customer\Controllers\ApplicationController;
+use GuzzleHttp\Client;
+
+class CustomerController extends ApplicationController
+{
+    private $params;
+    private $decoded;
+    private $index_entity;
+    private $customerService;
+    private $customer;
+
+    public function __construct($params, $decoded, $index_entity)
+    {
+        $this->params = $params;
+        $this->decoded = $decoded;
+        $this->index_entity = $index_entity;
+        $this->customerService = new CustomerService();
+    }
+    
+    public function processRequest()
+    {
+        if (strcasecmp($_SERVER['REQUEST_METHOD'], 'POST') === 0) {
+            //Signup a customer
+            $this->signupCustomer(
+                $this->decoded['firstname'],
+                $this->decoded['lastname'],
+                $this->decoded['phonenumber'],
+                $this->decoded['email'],
+                $this->decoded['gender'],
+                $this->decoded['country'],
+                $this->decoded['dateofbirth']
+            );
+        } elseif (strcasecmp($_SERVER['REQUEST_METHOD'], 'GET')
+            === 0 && sizeof($this->params) === $this->index_entity+2) {
+            //retrieve customer information by id
+            $customerId = $this->params[sizeof($this->params)-1];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function sizeof() is discouraged; use count() instead.](https://app.codacy.com/gh/BuildForSDG/Team-064-Backend/pullRequest?prid=5616370)